### DISCUSSION TITLE
v4.2.1

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#11396] [General] [answered]
### Release Notes Highlights
Requires: **.NET 6.0 Desktop Runtime [v6.0.25](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-6.0.25-windows-x64-installer)** or later

#### Changes:
* [#11393] Avoid popup resetting new file to index
* [#11389] GPG: don't crash on artificial commits
* [#11384] Do not migrate empty settings
* [#11381] GitExtSshAskPass: Ensure chars are outputed as UTF8 chars
* [#11376] Delete unused branch plugin: Makes GetBranchNames() support worktrees
* [#11369] Let git deny the commit if committer is not specified

[#11393]:https://github.com/gitextensions/gitextensions/pull/11393
[#11389]:https://github.com/gitextensions/gitextensions/pull/11389
[#11384]:https://github.com/gitextensions/gitextensions/pull/11384
[#11381]:https://github.com/gitextensions/gitextensions/pull/11381
[#11376]:https://github.com/gitextensions/gitextensions/pull/11376
[#11369]:https://github.com/gitextensions/gitextensions/pull/11369

Full change log: https://github.com/gitextensions/gitextensions/blob/v4.2.1/GitUI/Resources/ChangeLog.md#version-421-28-nov-2023

![Downloads](https://img.shields.io/github/downloads/gitextensions/gitextensions/v4.2.1/total.svg)

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v4.2.1'>v4.2.1</a>.</em>

**Comment by m-karen** [comment]
Why is the ugly GUI bug back, where you can't stretch the contents of the window to the edge on the right, and there remains an unnecessary and unused piece of the window? See the red arrow in the attached screenshot:
![screenshot](https://github.com/gitextensions/gitextensions/assets/133285564/6da4d008-130a-44d1-87ee-8dd2a093d53b)
It tears my eyes out. :-( It was there first I guess in version "_GitExtensions-4.1.0.16698-1fe52a137_", then in version "_GitExtensions-Portable-4.2.0.17069-5a11dfdf5_" it was fixed and now it's there again.

**Reply by RussKie** [reply, marked as answer]
No idea. But you can make it disappear cycling through these:
![image](https://github.com/gitextensions/gitextensions/assets/4403806/d767b645-b1a3-4cae-b3f3-8f6d3de56336)

/cc: @mstv

**Reply by m-karen** [reply]
Wow, thank You very much for the **Commit info below graph** hint!

**Comment by chirontt** [comment]
A manual build of this v4.2.1 version for Windows on Arm64 is available in my fork: https://github.com/chirontt/gitextensions/releases/tag/v4.2.1-win32-aarch64

**Comment by jeveno** [comment]
The download link doesn't work

**Reply by m-karen** [reply]
The link by chirontt works normally for me right now.

**Reply by RussKie** [reply]
The links are working. 
Could you please be more specific which link isn't working for you?

**Comment by Harshith1b** [comment]
I have installed this version but while launching it I'm getting this issue its only asking for version 6.0.25 but I'm unable to install v6.0.25.
I'm allowed to install only v8 or above I tried it by installing v8 and launching git extension but I'm getting popup only v6 is supported.  but in this page, its saying .NET 6.0 Desktop Runtime [v6.0.25] or later is there any solution for this????

**Reply by RussKie** [reply]
In theory, you could try to modify "C:\Program Files (x86)\GitExtensions\GitExtensions.runtimeconfig.json" and "C:\Program Files (x86)\GitExtensions\BugReporter.runtimeconfig.json" to something like this, but no guarantee of any kind it will work:
```diff
 {
     "runtimeOptions": {
-        "tfm": "net6.0",
+        "tfm": "net8.0",
         "frameworks": [
             {
                 "name": "Microsoft.WindowsDesktop.App",
-                "version": "6.0.25"
+                "version": "8.0.0"
             }
         ],
         "configProperties": {
             "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
         }
     }
 }
```

The version you can get from `dotnet -info`

**Reply by RussKie** [reply]
Other than that, your best bet is to use the latest dev version which is built against .NET 8 --> https://ci.appveyor.com/project/gitextensions/gitextensions/branch/master/artifacts

**Reply by Harshith1b** [reply]
Thank you
